arch/x86_64/mcount-dynamic.c

```diff
@@ -151,12 +153,13 @@ static void read_patchable_loc(struct mcount_dynamic_info *mdi, struct uftrace_e
 	elf_read_secdata(elf, iter, 0, mdi->patch_target, shdr->sh_size);
 
 	/* symbol has relative address, fix it to match each other */
-	if (elf->ehdr.e_type == ET_EXEC) {
+	if (elf->ehdr.e_type == ET_EXEC || elf->ehdr.e_type == ET_DYN) {
 		unsigned long *patchable_loc = mdi->patch_target;
 		unsigned i;
 
 		for (i = 0; i < mdi->nr_patch_target; i++) {
-			patchable_loc[i] -= offset;
+			if (elf->ehdr.e_type == ET_EXEC)
+				patchable_loc[i] -= offset;
```

Comment:
Hmm.. what's the point of this change?  I think it already supports `ET_DYN`, no?

Reply:
Right, this change is not needed. I've just dropped it.